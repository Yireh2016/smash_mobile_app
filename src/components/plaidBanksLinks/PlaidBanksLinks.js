import * as React from 'react';
import PlaidLink from 'react-native-plaid-link-sdk';
import PropTypes from 'prop-types';

//TODO: estos comentarios son para cuando plaid link este en la version 7 que
//tiene soporte para typescript
// export interface PlaidBanksLinksInterface {
//   onExit: LinkExitListener;
//   onSuccess: (data: any) => void;
//   tokenConfig: LinkTokenConfiguration;
//   children?: Element;
// }

const PlaidBanksLinks = ({ onSuccess, tokenConfig, onExit, children }) => {
  return (
    <>
      <PlaidLink
        clientName="Smash"
        token={tokenConfig.token}
        onSuccess={(data) => {
          onSuccess(data);
        }}
        onExit={onExit}
      >
        {children}
      </PlaidLink>
    </>
  );
};

PlaidBanksLinks.propTypes = {
  onExit: PropTypes.func,
  onSuccess: PropTypes.func,
  tokenConfig: PropTypes.object,
  children: PropTypes.element,
};

export default PlaidBanksLinks;
