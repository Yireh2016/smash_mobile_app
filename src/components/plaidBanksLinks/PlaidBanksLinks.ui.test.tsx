import * as React from 'react';
import { Text } from 'react-native-svg';
import renderer from 'react-test-renderer';
import PlaidBanksLinks from './PlaidBanksLinks';

describe('PlaidBanksLinks', () => {
  it('PlaidBanksLinks default', () => {
    const ComponentProps = {
      onSuccess: () => jest.fn(),
      tokenConfig: { token: 'test' },
      onExit: () => jest.fn(),
    };
    const Element = () => {
      return (
        <>
          {ComponentProps.tokenConfig.token === '' ? null : (
            <PlaidBanksLinks {...ComponentProps}>
              <Text>test</Text>
            </PlaidBanksLinks>
          )}
        </>
      );
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('PlaidBanksLinks without token', () => {
    const ComponentProps = {
      onSuccess: () => jest.fn(),
      tokenConfig: { token: '' },
      onExit: () => jest.fn(),
    };
    const Element = () => {
      return (
        <>
          {ComponentProps.tokenConfig.token === '' ? null : (
            <PlaidBanksLinks {...ComponentProps}>
              <Text>test</Text>
            </PlaidBanksLinks>
          )}
        </>
      );
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
