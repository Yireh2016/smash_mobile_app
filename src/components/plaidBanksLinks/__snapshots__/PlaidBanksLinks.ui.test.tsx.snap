// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaidBanksLinks PlaidBanksLinks default 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <RNSVGText
    content={null}
    dx={Array []}
    dy={Array []}
    font={Object {}}
    rotate={Array []}
    x={Array []}
    y={Array []}
  >
    <RNSVGTSpan
      content="test"
      dx={Array []}
      dy={Array []}
      font={Object {}}
      rotate={Array []}
      x={Array []}
      y={Array []}
    />
  </RNSVGText>
</View>
`;

exports[`PlaidBanksLinks PlaidBanksLinks without token 1`] = `null`;
